hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/AbstractMapSetWithMaxIdleMessageTask.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.map;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.instance.Node;
+import com.hazelcast.internal.cluster.Versions;
+import com.hazelcast.nio.Connection;
+
+public abstract class AbstractMapSetWithMaxIdleMessageTask<P> extends AbstractMapSetMessageTask<P> {
+
+    protected AbstractMapSetWithMaxIdleMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected void beforeProcess() {
+        super.beforeProcess();
+        checkCompatibility();
+    }
+
+    void checkCompatibility() {
+        if (nodeEngine.getClusterService().getClusterVersion().isLessThan(Versions.V3_11)) {
```

Comment:
Rolling upgrade from older than 1 previous version is not supported. Cluster version cannot be less than 3.11. There is no need for this check.

Reply:
Ok. Removed it. Thanks.